x-pack/plugins/actions/server/routes/update.test.ts

```diff
@@ -0,0 +1,214 @@
+ */
+
+// it('calls the update function with proper parameters', async () => {
+//   const request = {
+//     method: 'PUT',
+//     url: '/api/action/1',
+//     payload: {
+//       name: 'My name',
+//       config: { foo: true },
+//     },
+//   };
+//   const updateResult: ActionResult = {
+//     id: '1',
+//     actionTypeId: 'my-action-type-id',
+//     name: 'My name',
+//     config: { foo: true },
+//   };
+
+//   actionsClient.update.mockResolvedValueOnce(updateResult);
+//   const { payload, statusCode } = await server.inject(request);
+//   expect(statusCode).toBe(200);
+//   const response = JSON.parse(payload);
+//   expect(response).toEqual({
+//     id: '1',
+//     actionTypeId: 'my-action-type-id',
+//     name: 'My name',
+//     config: { foo: true },
+//   });
+//   expect(actionsClient.update).toHaveBeenCalledTimes(1);
+//   expect(actionsClient.update.mock.calls[0]).toMatchInlineSnapshot(`
+//     Array [
+//       Object {
+//         "action": Object {
+//           "config": Object {
+//             "foo": true,
+//           },
+//           "name": "My name",
+//           "secrets": Object {},
+//         },
+//         "id": "1",
+//       },
+//     ]
+//   `);
+// });
+
+import { updateActionRoute } from './update';
+import { mockRouter, RouterMock } from '../../../../../src/core/server/http/router/router.mock';
+import { mockLicenseState } from '../lib/license_state.mock';
+import { verifyApiAccess } from '../lib/license_api_access';
+import { mockHandlerArguments } from './_mock_handler_arguments';
+
+jest.mock('../lib/license_api_access.ts', () => ({
+  verifyApiAccess: jest.fn(),
+}));
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('updateActionRoute', () => {
+  it('updates an action with proper parameters', async () => {
+    const licenseState = mockLicenseState();
+    const router: RouterMock = mockRouter.create();
+
+    updateActionRoute(router, licenseState);
+
+    const [config, handler] = router.put.mock.calls[0];
+
+    expect(config.path).toMatchInlineSnapshot(`"/api/action/{id}"`);
+    expect(config.options).toMatchInlineSnapshot(`
+      Object {
+        "tags": Array [
+          "access:actions-all",
+        ],
+      }
+    `);
+
+    const updateResult = {
+      id: '1',
+      actionTypeId: 'my-action-type-id',
+      name: 'My name',
+      config: { foo: true },
+    };
+
+    const actionsClient = {
+      update: jest.fn().mockResolvedValueOnce(updateResult),
+    };
+
+    const [context, req, res] = mockHandlerArguments(
+      { actionsClient },
+      {
+        params: {
+          id: '1',
+        },
+        body: {
+          name: 'My name',
+          config: { foo: true },
+          secrets: { key: 'i8oh34yf9783y39' },
+        },
+      },
+      ['ok']
+    );
+
+    expect(await handler(context, req, res)).toEqual({ body: updateResult });
+
+    expect(verifyApiAccess).toHaveBeenCalledWith(licenseState);
```

Comment:
same with a previous one; not sure we need the test below this one, if it's primary purpose is only to run this `expect()`.